packages/pie/src/PieCanvas.tsx

```diff
@@ -58,61 +75,60 @@ const drawRadialLabels = (ctx, labels, theme, linkStrokeWidth) => {
     })
 }
 
-const PieCanvas = ({
+// prettier-ignore
+const PieCanvas = <R, >({
     data,
-    id,
-    value,
+    id = defaultProps.id,
+    value = defaultProps.value,
     valueFormat,
-    sortByValue,
-
-    // layers,
+    sortByValue = defaultProps.sortByValue,
 
-    startAngle,
-    endAngle,
-    padAngle,
-    fit,
-    innerRadius: innerRadiusRatio,
-    cornerRadius,
+    startAngle = defaultProps.startAngle,
+    endAngle = defaultProps.endAngle,
+    padAngle = defaultProps.padAngle,
+    fit = defaultProps.fit,
+    innerRadius: innerRadiusRatio = defaultProps.innerRadius,
+    cornerRadius = defaultProps.cornerRadius,
 
     width,
     height,
     margin: partialMargin,
-    pixelRatio,
+    pixelRatio = 1,
 
-    colors,
+    colors = defaultProps.colors,
 
     // border
-    borderWidth,
-    borderColor,
+    borderWidth = defaultProps.borderWidth,
+    borderColor = defaultProps.borderColor as InheritedColorProp<ComputedDatum<R>>,
 
     // radial labels
-    radialLabel,
-    enableRadialLabels,
-    radialLabelsSkipAngle,
-    radialLabelsLinkOffset,
-    radialLabelsLinkDiagonalLength,
-    radialLabelsLinkHorizontalLength,
-    radialLabelsLinkStrokeWidth,
-    radialLabelsTextXOffset,
-    radialLabelsTextColor,
-    radialLabelsLinkColor,
+    radialLabel = defaultProps.radialLabel,
+    enableRadialLabels = defaultProps.enableRadialLabels,
+    radialLabelsSkipAngle = defaultProps.radialLabelsSkipAngle,
+    radialLabelsLinkOffset = defaultProps.radialLabelsLinkOffset,
+    radialLabelsLinkDiagonalLength = defaultProps.radialLabelsLinkDiagonalLength,
+    radialLabelsLinkHorizontalLength = defaultProps.radialLabelsLinkHorizontalLength,
+    radialLabelsLinkStrokeWidth = defaultProps.radialLabelsLinkStrokeWidth,
+    radialLabelsTextXOffset = defaultProps.radialLabelsTextXOffset,
+    radialLabelsTextColor = defaultProps.radialLabelsTextColor,
+    radialLabelsLinkColor = defaultProps.radialLabelsLinkColor,
 
     // slices labels
-    sliceLabel,
-    enableSliceLabels,
-    sliceLabelsSkipAngle,
-    sliceLabelsTextColor,
-    sliceLabelsRadiusOffset,
+    sliceLabel = defaultProps.sliceLabel,
+    enableSliceLabels = defaultProps.enableSliceLabels,
+    sliceLabelsSkipAngle = defaultProps.sliceLabelsSkipAngle,
+    sliceLabelsTextColor = defaultProps.sliceLabelsTextColor,
+    sliceLabelsRadiusOffset = defaultProps.sliceLabelsRadiusOffset,
 
     // interactivity
-    isInteractive,
+    isInteractive = defaultProps.isInteractive,
     onClick,
     onMouseMove,
-    tooltip,
+    tooltip = defaultProps.tooltip,
 
-    legends,
-}) => {
-    const canvasEl = useRef(null)
+    legends = defaultProps.legends,
+}: PieCanvasProps<R>) => {
+    const canvasEl = useRef<HTMLCanvasElement>(null)
```

Comment:
```suggestion
    const canvasEl = useRef<HTMLCanvasElement | null>(null)
```

This is the proper type since we initialize to `null` here.

Reply:
Adjusted.